src/Illuminate/Validation/Validator.php

```diff
@@ -527,6 +530,37 @@ protected function validateSometimes()
     }
 
     /**
+     * "Break" on first fail.
+     *
+     * Always returns true, just lets us put failonfirst in rules.
+     *
+     * @return bool
+     */
+    protected function validateFailonfirst()
+    {
+        return true;
+    }
+
+    /**
+     * Stop on error if failonfirst rule is given.
+     *
+     * First check if 'failonfirst' is set,
+     * if set return true on validation fail.
+     *
+     * @return bool
+     */
+    protected function shouldBreakOnFail($attribute)
+    {
+        if ($this->hasRule($attribute, ['Failonfirst'])) {
+            if ($this->messages->has($attribute)) {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
@lucasmichot That is good but I think the previous one is fast and easily readable? Should I change?